src/main/java/org/folio/service/PatronBlocksService.java

```diff
@@ -79,10 +79,6 @@ public PatronBlocksService(Map<String, String> okapiHeaders, Vertx vertx) {
   private AutomatedPatronBlocks calculateBlocks(BlocksCalculationContext ctx) {
     final AutomatedPatronBlocks blocks = new AutomatedPatronBlocks();
 
-    if (ctx.patronBlockLimits.isEmpty()) {
-      return blocks;
-    }
-
```

Comment:
Maybe, instead of deleting this part, we should add it to all of the steps after fetching the limits? Also, to avoid hardcoding of this condition in every method I personally would create a method in context, something like `boolean shouldCalculationBeSkipped()`. That would allow to create complex conditions to determine if calculation is pointless in the future.

Reply:
instead, I extracted the part where limits should be non empty, if it is not good approach, I will change as you suggested here

Reply:
Hm.. I would do it like that for two reasons:
- We're limiting ourselves to 'all or nothing' approach. In the future I expect we'll have more fine-grained logic. For example, if 'Max fee/fine balance' limit is missing, we're not even going to try to calculate if it's exceeded, etc. With the approach I proposed we'll be able to add more `..shouldBeSkipped..` methods to the context and skip smaller parts of the workflow.
- Breaking the chain in pieces makes it harder to read and understand what is done by the service

Reply:
ok, understood. Thanks a lot for good explanation!

Reply:
fixed